packages/oauth/src/stores/file-store.ts

```diff
@@ -0,0 +1,116 @@
+import { Installation, InstallationStore, InstallationQuery } from '../index';
+import { Logger } from '../logger';
+import fs from 'fs';
+import path from 'path';
+import { homedir } from 'os';
+
+export interface FileInstallationOptions {
+  baseDir?: string;
+  historicalDataEnabled?: boolean;
+  clientId?: string;
+}
+
+export class FileInstallationStore implements InstallationStore {
+
+  private baseDir: string;
+  private historicalDataEnabled: boolean;
+
+  constructor({
+    baseDir = `${homedir()}/.bolt-app-installation`,
+    clientId,
+    historicalDataEnabled = true,
+  }: FileInstallationOptions = {}) {
+    this.baseDir = clientId !== undefined ? `${baseDir}/${clientId}` : baseDir;
+    this.historicalDataEnabled = historicalDataEnabled;
+  }
+
+  public async storeInstallation(installation: Installation, logger?: Logger): Promise<void> {
+    const { enterprise, team, user } = installation;
+    const installationData = JSON.stringify(installation);
+    const installationDir = this.getInstallationDir(enterprise?.id, team?.id);
+
+    if (logger !== undefined) { logger.warn('Storing installation in FileInstallationStore'); }
+
+    // Create Installation Directory
+    fs.mkdir(installationDir, { recursive: true }, (err) => {
+      if (err !== null) { return console.error(err); }
+    });
+
+    try {
+      writeToFile(`${installationDir}/app-latest`, installationData);
+      writeToFile(`${installationDir}/user-${user.id}-latest`, installationData);
+
+      if (this.historicalDataEnabled) {
+        const currentUTC = Date.now();
+        writeToFile(`${installationDir}/app-${currentUTC}`, installationData);
+        writeToFile(`${installationDir}/user-${user.id}-${currentUTC}`, installationData);
+      }
+    } catch (err) {
+      throw new Error('Failed to save installation to FileInstallationStore');
+    }
+  }
+
+  public async fetchInstallation(query: InstallationQuery<boolean>, logger?: Logger): Promise<Installation> {
+    const { enterpriseId, teamId } = query;
```

Comment:
`MemoryInstallationStore` doesn't take into consideration `user_id`, so I omitted it here, but now that I'm reading over it in this context.. 

Given that `user.id` has been introduced in `storeInstallation` above, my gut feeling is that it needs to be introduced here, as well, but confirmation of this from others would be useful.

Reply:
How about having TODO comments on user token scope installation support?

Reply:
@seratch so to support user tokens, it seems we would want to fetch `/user-{user_id}-latest` from the installationDirectory. How do you determine if you want to return this over `app-latest`? Do you look for the existence of `user_id` in the `query` and  `/user-{user_id}-latest` on the file system and return it over `app-latest`? How about if you were explicitly looking for a user token existing, in that scenario if `/user-{user_id}-latest` didn't exist, you would want to throw an error instead of getting `app-latest` which might contain a user token belonging to a different user.

Reply:
@stevengill Performing an additional query would be necessary for it. Here are the ways we do in Python / Java.

* https://github.com/slackapi/bolt-python/blob/v1.6.1/slack_bolt/authorization/authorize.py#L137-L167
* https://github.com/slackapi/java-slack-sdk/blob/v1.8.1/bolt/src/main/java/com/slack/api/bolt/middleware/builtin/MultiTeamsAuthorization.java#L128-L138

Reply:
@seratch this is very helpful! So the idea being do two fetchinstallation calls in authorize (or do this logic in fetchInstallatino itself). Roughly, in this JS implementation, the first fetchInstallation call would grab `app-latest`. We would want to strip out `user_id` from query so it doesn't try to grab a `user-{user_id}-latest` instead. The second call to fetchInstallation we would pass in the userId and it would look for `user-{user_id}-latest`. It would pull out the user portion of the installation Object and attach it to the original installationObject we get back from `app-latest`. It covers some of the edge cases I was worried about. Nice solution.

Reply:
To limit the scope, I'm in favor of leaving a TODO to address the user token support at a later time if others are.